services/asset/httpimpl/http.go

```diff
@@ -453,3 +479,44 @@ func customLoggerMiddleware(logger ulogger.Logger) echo.MiddlewareFunc {
 		}
 	}
 }
+
+// initGRPCClients initializes persistent gRPC connections to P2P and BlockValidation services.
+// Uses lazy initialization with sync.Once to ensure connections are only created once.
+// These persistent connections automatically reconnect on failure (using grpc.NewClient).
+func (h *HTTP) initGRPCClients(ctx context.Context) {
```

Comment:
This should not be here. The clients should be created and passed in from the daemon